### PR TITLE
Added a cypress test to assign a volunteer to a patient

### LINKED ISSUE #8913: Add a cypress test to assign a volunteer to a patient
**Description**

Add a new cypress test to assign a volunteer to an existing live patient

**General Note**

1. Follow the Page Object Model (POM) approach. : [reference](https://github.com/ohcnetwork/care_fe/blob/develop/cypress/e2e/patient_spec/PatientLogUpdate.cy.ts) 
2. Refer to commands.ts in the Cypress folder to understand existing reusable components.
3. Use proper function names for easier code review.
4.Since there's nothing related to volunteers currently, create a new file named PatientVolunteer.cy.ts under the patient_spec folder.

**Steps to Assign a volunteer**

1. Go to the patient tab
2. click on any live patient card
3. On the patient consultation page, click the Patient Details button in the top-right corner
4. Once the new page loads, scroll down to the Assign New Volunteer button.
5. Assign a new volunteer to the patient.
6. Ensure the newly added volunteer's name is displayed as a banner at the top of the patient details page.

**Make sure to verify the below things**

1. Ensure the normal workflow is functional.
2. Replace the existing volunteer with a new one and verify the banner shows the new volunteer's name.
3. When the volunteer is removed and left blank, verify that the banner is no longer displayed.

### PR DESCRIPTION
**Pull request by sidpg123**
## Proposed Changes

- Fixes #8913 
- Change 1:  Created a new file named PatientVolunteer.cy.ts under the patient_spec folder for testing purposes.
- Change 2: Created PatientDetails.ts for the Patient Details page functionality.
- Change 3: Added id attribute to the banner element for better targeting in tests.

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Introduced a test suite for assigning and unassigning volunteers to patients, ensuring the UI reflects changes accurately.
	- Added a unique identifier for the assigned volunteer display in the PatientHome component to enhance accessibility.
	- Enhanced the Autocomplete component with `data-testid` attributes for improved testability.

- **Tests**
	- Implemented methods for interacting with the patient details page, including volunteer assignment and verification of UI updates.
	- Added error handling tests for scenarios where a volunteer is not found in the dropdown.

### CODE REVIEW

File: cypress/e2e/patient_spec/PatientVolunteer.cy.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    patientDetailsPage.selectAndAssignVolunteer(anotherVolunteerName);
+
+    patientDetailsPage.verifyVolunteerBannerIsUpdated(anotherVolunteerName);
+
+    patientDetailsPage.clickAssignToAVounteer();
+
+    patientDetailsPage.unassignVolunteer();
+
+    patientDetailsPage.verifyBannerIsRemovedAfterUnassign();
+  });
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply cypress/e2e/patient_spec/PatientVolunteer.cy.ts]: Some of the suggestions are worth doing, like this one (break it up into smaller tests). Go ahead and make the changes accordingly @sidpg123

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

**Comment by sidpg123** [comment]
@Jacobjeevan i will do it.

**Comment by Jacobjeevan** [comment]
@sidpg123 PatientHome just changed, so take a look (solve the merge conflicts and rerun the tests locally).

**Comment by sidpg123** [comment]
@Jacobjeevan ok I will do it.

**Comment by nihal467** [comment]
@sidpg123 what is the status on this PR

**Comment by sidpg123** [comment]
@nihal467 All requirements have been fulfilled. I have made all changes requested by coderabbitai. If any further changes are required, please let me know.

**Review by sidpg123 — COMMENTED** [review]
All requested changes are completed.

File: cypress/e2e/patient_spec/PatientVolunteer.cy.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  const patientPage = new PatientPage();
+  const patientConsultationPage = new PatientConsultationPage();
+  const patientDetailsPage = new PatientDetailsPage();
+  const patient = "Dummy Patient 16";
+  const volunteerName = "dummy volunteer";
+  const anotherVolunteerName = "Abhi Patil";
+
+  before(() => {
+    cy.log("Logging in as district admin");
+    loginPage.loginAsDistrictAdmin();
```
Comment (nihal467) [inline cypress/e2e/patient_spec/PatientVolunteer.cy.ts]:
```suggestion
    loginPage.loginByRole("districtAdmin");
```

File: cypress/e2e/patient_spec/PatientVolunteer.cy.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  const loginPage = new LoginPage();
+  const patientPage = new PatientPage();
+  const patientConsultationPage = new PatientConsultationPage();
+  const patientDetailsPage = new PatientDetailsPage();
+  const patient = "Dummy Patient 16";
+  const volunteerName = "dummy volunteer";
+  const anotherVolunteerName = "Abhi Patil";
+
+  before(() => {
+    cy.log("Logging in as district admin");
```
Comment (nihal467) [inline cypress/e2e/patient_spec/PatientVolunteer.cy.ts]:
```suggestion
    cy.log("Logging in as district admin");
```
remove unwanted logging

File: cypress/e2e/patient_spec/PatientVolunteer.cy.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    cy.wrap(null, { timeout: 10000 }).then(() => {
+      patientPage.visitPatient(patient);
+    });
+
+    // Verify patient details page is accessible
+    cy.get("#patient-details").should("exist");
+    patientConsultationPage.clickPatientDetails();
+  });
+
+  describe("volunteer assignment workflow", () => {
```
Comment (nihal467) [inline cypress/e2e/patient_spec/PatientVolunteer.cy.ts]:
```suggestion
  describe("volunteer assignment workflow", () => {
```

Combine everything into one single test

File: cypress/pageobject/Patient/PatientDetails.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
+export class PatientDetailsPage {
+  clickAssignToVolunteer() {
+    cy.contains("button", "Assign to a Volunteer")
```
Comment (nihal467) [inline cypress/pageobject/Patient/PatientDetails.ts]:
```suggestion
    cy.contains("button", "Assign to a Volunteer")
```

use proper Id for the button

Comment (nihal467) [reply cypress/pageobject/Patient/PatientDetails.ts]: check for cy.verifyAndClickElement() function after you give them the id

File: cypress/pageobject/Patient/PatientDetails.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+      .scrollIntoView()
+      .should("be.visible")
+      .should("be.enabled")
+      .click();
+  }
+
+  selectAndAssignVolunteer(volunteerName: string) {
+    cy.clickAndSelectOption("#assign_volunteer", volunteerName);
+    cy.clickSubmitButton("Assign");
+    cy.get("#assigned-volunteer", { timeout: 10000 })
```
Comment (nihal467) [inline cypress/pageobject/Patient/PatientDetails.ts]: check for cy.verifyAndClickElement() function

File: cypress/pageobject/Patient/PatientDetails.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      .scrollIntoView()
+      .should("be.visible")
+      .should("contain.text", volunteerName);
+  }
+
+  verifyVolunteerBannerIsUpdated(volunteerName: string) {
+    cy.get("#assigned-volunteer").should(
```
Comment (nihal467) [inline cypress/pageobject/Patient/PatientDetails.ts]:
```suggestion
    cy.get("#assigned-volunteer").should(
```

check the function cy.verifyContentPresence

**Review by nihal467 — CHANGES_REQUESTED** [review]
- check the existing command.ts file, helper functions in the cypress and modify everything properly 

https://docs.ohc.network/docs/care/development/guidelines-to-write-cypress/

**Comment by nihal467** [comment]
@sidpg123 Could you please provide an update on the status of this PR? If there’s no response within the next 48 hours, we will proceed to close it due to inactivity.

**Comment by Jacobjeevan** [comment]
@sidpg123 What's the status on this PR?

**Comment by rithviknishad** [comment]
no longer relevant in new architecture

**Pull request closed without merging**